domain/user/state/state.go

```diff
@@ -291,6 +291,95 @@ func (st *State) SetPasswordHash(ctx context.Context, uuid user.UUID, passwordHa
 	})
 }
 
+// EnableUserAuthentication will enable the user with the supplied UUID. If the user does not
+// exist an error that satisfies usererrors.NotFound will be returned.
+func (st *State) EnableUserAuthentication(ctx context.Context, uuid user.UUID) error {
+	db, err := st.DB()
+	if err != nil {
+		return errors.Annotate(err, "getting DB access")
+	}
+
+	enableUserQuery := `
+UPDATE user_authentication
+SET disabled = false
+WHERE user_uuid = $M.uuid
+`
+
+	updateEnableUserStmt, err := sqlair.Prepare(enableUserQuery, sqlair.M{})
+	if err != nil {
+		return errors.Annotate(err, "preparing update enableUserAuthentication query")
+	}
+
+	return db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		removed, err := st.isRemoved(ctx, tx, uuid)
+		if err != nil {
+			return errors.Annotatef(err, "getting user with uuid %q", uuid)
+		}
+		if removed {
+			return errors.Annotatef(usererrors.NotFound, "%q", uuid)
+		}
+
+		err = ensureUserAuthentication(ctx, tx, uuid)
```

Comment:
I am overall fine with doing is this way. But if you want to get extra slick about the implementation we can do it all in one statement.

If you change the UPDATE sql above to an insert then write the ON CONFLICT handler to do an upsert instead you can avoid the need to go back and forth to the DB twice.